File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/06. Easter Competition.py
count_easter_bread = int(input())
name_backer = None
current_score = 0
max_score = 0

max_score_name = None

for i in range(1, count_easter_bread + 1):
    name_backer = input()
    while True:
        command = input()
        if command == "Stop":
            break
        current_score += int(command)
    print(f"{name_backer} has {current_score} points.")
    if current_score > max_score:
        print(f"{name_backer} is the new number 1!")
        max_score = current_score
        max_score_name = name_backer
    current_score = 0

print(f"{max_score_name} won competition with {max_score} points!")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/06. Easter Competition.js
function easterCompetition(input) {
    // Mask
    let count = Number(input.shift());

    let bestChef = undefined;
    let bestPoints = Number.MIN_SAFE_INTEGER;

    for (let times = 1; times <= count; times += 1) {
        let chef = input.shift();
        let points = 0;
        while ('command is Stop') {
            let command = input.shift();
            if (command == "Stop") {
                console.log(`${chef} has ${points} points.`);
                if (points > bestPoints) {
                    bestChef = chef;
                    bestPoints = points;
                    console.log(`${chef} is the new number 1!`);
                }
                break;
            }
            points += Number(command);

        }
    }

    console.log(`${bestChef} won competition with ${bestPoints} points!`);
}

easterCompetition([3, 'Chef Manchev', 10, 10, 10, 10, 'Stop', 'Natalie', 8, 2, 9, 'Stop', 'George', 9, 2, 4, 2, 'Stop']); 
// Should return:
// Chef Manchev has 40 points.
// Chef Manchev is the new number 1!
// Natalie has 19 points.
// George has 17 points.
// Chef Manchev won competition with 40 points!


easterCompetition([2, 'Chef Angelov', 9, 9, 9, 'Stop', 'Chef Rowe', 10, 10, 10, 10, 'Stop']);  
// Should return:
// Chef Angelov has 27 points.
// Chef Angelov is the new number 1!
// Chef Rowe has 40 points.
// Chef Rowe is the new number 1!
// Chef Rowe won competition with 40 points!
